freqtrade/freqai/data_drawer.py

```diff
@@ -163,37 +164,67 @@ def load_historic_predictions_from_disk(self):
         Locate and load a previously saved historic predictions.
         :return: bool - whether or not the drawer was located
         """
-        exists = self.historic_predictions_path.is_file()
+        exists = self.historic_predictions_folder.exists()
+        convert = self.historic_predictions_path.is_file()
+
         if exists:
             try:
-                with self.historic_predictions_path.open("rb") as fp:
-                    self.historic_predictions = cloudpickle.load(fp)
+                self.load_historic_predictions_from_folder()
                 logger.info(
                     f"Found existing historic predictions at {self.full_path}, but beware "
                     "that statistics may be inaccurate if the bot has been offline for "
                     "an extended period of time."
                 )
             except EOFError:
                 logger.warning(
-                    'Historical prediction file was corrupted. Trying to load backup file.')
-                with self.historic_predictions_bkp_path.open("rb") as fp:
-                    self.historic_predictions = cloudpickle.load(fp)
-                logger.warning('FreqAI successfully loaded the backup historical predictions file.')
+                    'Historical prediction files were corrupted. Trying to load backup files.')
+                self.load_historic_predictions_from_folder()
+                logger.warning('FreqAI successfully loaded the backup '
+                               'historical predictions files.')
+
+        elif not exists and convert:
+            logger.info("Converting your historic predictions pkl to parquet"
+                        "to improve performance.")
+            with Path.open(self.historic_predictions_path, "rb") as fp:
+                self.historic_predictions = cloudpickle.load(fp)
+            self.save_historic_predictions_to_disk()
```

Comment:
would/should we delete the backup file after this step?
it's no longer written to - so once it's saved in the new format, it would stick around "forever" ... ? 
potentially even causing problems if the parquet files were reset - but this file forgotten ?

Reply:
Yeah good point, but I’m scared to do this as it is a very valuable file. Maybe we delete the main pkl but we leave the backup pkl, since the backup pkl won’t interfere with future operations.

Reply:
probably yes - the way it is now i think it's quite dangerous that (for whatever reason) a new migration might happen - which would overwrite the new (more precious as containing more data) files ...